src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsTypedHandler.java

```diff
@@ -151,14 +168,18 @@ private static boolean inStringLiteral(@NotNull PsiFile file,
     private static boolean inComment(@NotNull PsiFile file,
                                      @NotNull CharSequence documentChars,
                                      int offset) {
-        boolean inComment = false;
+        // Try to determine this using semantic tokens
```

Comment:
If the language server provides comment semantic tokens, we can use that to determine that the offset is within a comment without having to search for it textually.